eva_submission/eload_validation.py

```diff
@@ -353,6 +358,93 @@ def _collect_validation_workflow_results(self, output_dir):
             })
         self.eload_cfg.set('validation', 'assembly_check', 'pass', value=total_error == 0)
 
+    def _detect_structural_variant(self):
+
+        # Initializing the list of boolean values to store the status of the VCFs denoting whether it has a SV or not
+        has_sv = []
+        vcf_files = self._get_vcf_files()
+        has_sv = self._detect_structural_variant_from_variant_lines(vcf_files, has_sv)
+        validator_output_files = []
+        for vcf_file in vcf_files:
+            vcf_name = os.path.basename(vcf_file)
+            validator_output_files.append(os.path.join(self._get_dir('vcf_check'), vcf_name + '.vcf_validator.txt'))
+        has_sv = self._detect_structural_variant_from_validator_output(has_sv, validator_output_files)
+        for index in range(len(vcf_files)):
+            self.eload_cfg.set('validation', 'structural_variant_check', 'files', os.path.basename(vcf_files[index]),
+                               value={'has_structural_variant': has_sv[index]})
+        self.eload_cfg.set('validation', 'structural_variant_check', 'pass', value=True)
+        return has_sv
+
+    def _detect_structural_variant_from_variant_lines(self, vcf_files, has_sv):
+        no_of_variant_lines = []
+        vcf_count = -1
+        # Ref: https://samtools.github.io/hts-specs/VCFv4.3.pdf (Pages: 6, 16)
+        symbolic_allele_pattern = re.compile("^<(DEL|INS|DUP|INV|CNV|BND|DUP:TANDEM|DEL:ME.*|INS:ME.*)>$")
```

Comment:
This would work. But, in that case, the ALT field validation won't be performed. For e.g. if the ALT field contains a value <DELL>, then it would be treated as a valid ALT field. However, if the regular expression that I have given is used, then it would not consider it as a valid ALT field and only <DEL> would be considered as a valid ALT field.

Reply:
However, if you want, I can change that

Reply:
I have made this change now